### ISSUE TITLE
2-8 linkage options not being shown. 

### ISSUE THREAD

**Issue by gitoliver** [GLYCAM-Web/website#96]
**Tool:**
Carb builder options page: DNeup5Aca2-8DNeup5Aca2-3DGlcpb1-OH

**Bug Description:**
The 2-8 linkage has multiple conformers labeled A-E. However these aren't permutations, you pick one of A, B, C, D or E for the whole linkage. I assume this never worked. Some info is being printed by gmml and it's calculating the number of structures correctly, but I need to go check what the website is receiving, because we need to differentiate between the linkages types.
Gmml test output for the above sequence:
Number of likely shapes for this sequence is 10
Number of possible shapes for this sequence is 10
Name: DNeu5Acpa2-3DGlcpb, LinkageIndex: 1, Res1: 3, Res2: 2
    LikelyRotamers: Phi, t, -g
Name: DNeu5Acpa2-8DNeu5Acpa, LinkageIndex: 2, Res1: 4, Res2: 3
    LikelyRotamers: Phi, A, B, C, D, E
    LikelyRotamers: Psi, A, B, C, D, E
    LikelyRotamers: Omg8, A, B, C, D, E
    LikelyRotamers: Omg7, A, B, C, D, E
    LikelyRotamers: Omg9, A, B, C, D, E
Fin

**Comment by gitoliver** [comment]
This is still a broken thing on the site. Users can build 2-8 structures and a default is shown, but you cannot select anything on the options page and are stuck:
DNeup5Aca2-8DNeup5Aca2-OME
https://glycam.org/cb/options/55f1b1e7-6c49-43c9-ab9b-5fe240bbf661/
![image](https://github.com/user-attachments/assets/09d09d5e-a401-48c6-8ed0-35f86b909172)

**Comment by danwentworthart** [comment]
I can confirm that the website is receiving the data you describe. At this point the need is to decide what this should look like. The table solution provided above is designed for up to 3 dihedrals, each with 3 rotamers. This dataset requires 5 sets of 5. I will look into it.

**Comment by danwentworthart** [comment]
It is more complex than that, because the other linkage wants the Phi options to have the normal 3 column layout. Tables may not be the way if each record doesn't fit the same pattern. I will ponder this.

**Comment by gitoliver** [comment]
Yes it's different. Each linkage needs a row with the 5 options. A, B, C, D, E. They must only pick one of those for each linkage. Have everything be A by default. In the future, I can see us having more or less conformers (e.g. A-F, or A-B)

**Comment by gitoliver** [comment]
Here is the target:
![Image](https://github.com/user-attachments/assets/c07800f7-087e-4c4a-a683-bc0a46ad6ac6)

**Comment by gitoliver** [comment]
The problem is that this is what gets passed up to the website:
{
                                        "indexOrderedLabel": "2",
                                        "linkageLabel": null,
                                        "linkageName": "DNeu5Acpa2-8DNeu5Acpa",
                                        "firstResidueNumber": "4",
                                        "secondResidueNumber": "3",
                                        "dihedralsWithOptions": [
                                            "Phi",
                                            "Psi",
                                            "Omg8",
                                            "Omg7",
                                            "Omg9"
                                        ],
                                        "possibleRotamers": [
                                            {
                                                "dihedralName": "Phi",
                                                "dihedralValues": [
                                                    "A",
                                                    "B",
                                                    "C",
                                                    "D",
                                                    "E"
                                                ]
                                            },
                                            {
                                                "dihedralName": "Psi",
                                                "dihedralValues": [
                                                    "A",
                                                    "B",
                                                    "C",
                                                    "D",
                                                    "E"
                                                ]
                                            },
                                            {
                                                "dihedralName": "Omg8",
                                                "dihedralValues": [
                                                    "A",
                                                    "B",
                                                    "C",
                                                    "D",
                                                    "E"
                                                ]
                                            },
                                            {
                                                "dihedralName": "Omg7",
                                                "dihedralValues": [
                                                    "A",
                                                    "B",
                                                    "C",
                                                    "D",
                                                    "E"
                                                ]
                                            },
                                            {
                                                "dihedralName": "Omg9",
                                                "dihedralValues": [
                                                    "A",
                                                    "B",
                                                    "C",
                                                    "D",
                                                    "E"
                                                ]
                                            }
                                        ],
But this is not the right format. There should be something unique for RotamerType::conformer, it should not come under possibleRotamers. It would just be something like:
conformers: A, B, C, D, E.

For now, given that this needs gems and gmml2 work (it's also reporting them as possible rotamers), I'm going to comment out the extra rotamers here: https://github.com/GLYCAM-Web/gmml2/blob/main/src/MolecularMetadata/GLYCAM/dihedralangledata.cpp
And just have it be one shape. That way the website won't get any options to display, and we can build a single, reasonable shape for 2-8.

**Comment by gitoliver** [comment]
Pushed to gmml2 dev branch. Once integrated into main it will get into test.glycam.org. In my local dev env I see good behaviour: building 
DNeup5Aca2-8DNeup5Aca2-OH
gets me all the way to the download page without any options:

![Image](https://github.com/user-attachments/assets/74511396-fd17-4ba9-98b5-b949439ee28d)

Changing status of this bug to enhancement now, as 2-8's are working reasonably well, and adding extra shapes is a bonus.

**Comment by gitoliver** [comment]
Closing as no current plans to add these these options. The behavior we have now is ok.
